Jint/Native/Array/ArrayInstance.cs

```diff
@@ -648,13 +648,15 @@ private void ConvertToSparse()
 
         internal void EnsureCapacity(uint capacity)
         {
-            if (capacity <= MaxDenseArrayLength && capacity > (uint) _dense.Length)
+            if (capacity > MaxDenseArrayLength || _dense is null || capacity <= (uint) _dense.Length)
```

Comment:
I understand that the capacity doubles?

Reply:
Capacity is decided by the caller, we can grow as asked if it hasn't flipped to be dictionary (sparse) already. Usually logic asks for double the size but for example splice knows what to expect and thus was signaling specific size when algorithm is at specific point. Same like using constructor parameter, can signal known size.